### PR TITLE
fix: now you can use free text to select an organization for tracking repos

### LINKED ISSUE #2874: feat: Make Workspace org importing not user specific.
I'd love to see all orgs searchable in this list. 

Screen Shot 2024-03-05 at 12 12 23 PM

## Problem

I cannot demo worskpaces by importing an organization with someone. I can only watch them do it.

## Solution

If there is a technical limitation I am unaware and happy to close as won't fix, but the import for orgs on repository insights allow me to type an org and import. I'd love to follow that strategy in workspaces. 

Screen Shot 2024-03-05 at 12 18 54 PM

### PR DESCRIPTION
**Pull request by nickytonline**
## Description

Now, you can search for organizations you don't belong to. If the org does not exist, the search result will not return anything. Existing user organizations remain available and filterable.

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

Fixes #2874

## Mobile & Desktop Screenshots/Recordings

[Loom recording for Non-User Org Repo Tracking working](https://www.loom.com/share/8188983204b5471cae29496eaabf3363?sid=c0049bb6-6d77-4e66-881b-9e689fa2a38d)

## Steps to QA

1. Go to a workspace you can edit
2. Go to the settings.
3. Click on the add repositories button.
4. Notice the user organizations are available right away.
5. Type in an organization you are not a part of.
6. Notice that if the organization does not exist, no search results appear.
7. If the organization exists, it will be the only organization avialble in the drop down.
8. Select the user entered organization.
9. The associated repositories for it load up.

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

**Review by bdougie — APPROVED** [review]
LGTM

File: components/Workspaces/TrackedRepoWizard/TrackedRepoWizard.tsx [hunk 1]
```diff
@@ -24,6 +24,18 @@ type TrackedReposStep =
   | "filterPastedRepos"
   | "pickOrgRepos";
 
+async function getCaseSensitiveOrgName(orgSearchTerm: string) {
+  const response = await fetch(`https://api.github.com/orgs/${orgSearchTerm}`);
+
+  if (response.status === 200) {
+    // The organization name is case sensitive even if our search was case insensitive.
+    // Return the case sensitive organization name so we can use it to get the organization's repositories.
+    const { login } = await response.json();
```
Comment (jpmcb) [inline components/Workspaces/TrackedRepoWizard/TrackedRepoWizard.tsx:33]: This is very surprising to me especially given how our API handles all logins / orgs / repo names as case insensitive. Any reason? Is it that the org "Name" is different from the actual org login (that we use in the API). I.e., `https://github.com/citi` is the same as `https://github.com/CITI` but the "name" is `Citi`?

**Review by jpmcb — APPROVED** [review]
Screenshot 2024-03-19 at 6 25 51 PM

---

File: components/Workspaces/TrackedRepoWizard/TrackedRepoWizard.tsx (hunk 1 above)
Comment (nickytonline) [reply components/Workspaces/TrackedRepoWizard/TrackedRepoWizard.tsx:33]: Yeah, this surprised me as well. It’s not in our API though. It’s a React hook that hits the GitHub API.

See https://github.com/open-sauced/app/blob/beta/lib/hooks/useGetOrgRepos.ts

Comment (nickytonline) [reply components/Workspaces/TrackedRepoWizard/TrackedRepoWizard.tsx:33]: It was a casing mismatch on the client-side when filtering client-side. All sorted out in 290a778b1e6ea3ccccf78c377e538ea3d676510b

File: components/Workspaces/TrackedRepoWizard/SelectOrgReposStep.tsx [hunk 2]
```diff
@@ -55,7 +55,9 @@ export const SelectOrgReposStep = ({
   const [filteredRepositories, setFilteredRepositories] = useState<Map<string, boolean>>(repositories);
 
   useEffect(() => {
-    setFilteredRepositories(new Map([...repositories.entries()].filter(([repo]) => repo.startsWith(organization))));
+    setFilteredRepositories(
+      new Map([...repositories.entries()].filter(([repo]) => repo.toLowerCase().startsWith(organization)))
```
Comment (nickytonline) [inline components/Workspaces/TrackedRepoWizard/SelectOrgReposStep.tsx:59]: Needed to filter lowercased. Prior to this fix, the orgs users were a part of had the correct casing so this wasn't necessary, but now that we allow for user inputted orgs to search for, this is required.

File: components/Workspaces/TrackedRepoWizard/SelectOrgReposStep.tsx [hunk 3]
```diff
@@ -55,7 +55,11 @@ export const SelectOrgReposStep = ({
   const [filteredRepositories, setFilteredRepositories] = useState<Map<string, boolean>>(repositories);
 
   useEffect(() => {
-    setFilteredRepositories(new Map([...repositories.entries()].filter(([repo]) => repo.startsWith(organization))));
+    const lowercasedOrg = organization.toLowerCase();
+
+    setFilteredRepositories(
+      new Map([...repositories.entries()].filter(([repo]) => repo.toLowerCase().startsWith(lowercasedOrg)))
```
Comment (nickytonline) [inline components/Workspaces/TrackedRepoWizard/SelectOrgReposStep.tsx:61]: Needed to filter lowercased. Prior to this fix, the orgs users were a part of had the correct casing so this wasn't necessary, but now that we allow for user inputted orgs to search for, this is required.

File: components/Workspaces/TrackedRepoWizard/TrackedRepoWizard.tsx (hunk 1 above)
Comment (jpmcb) [reply components/Workspaces/TrackedRepoWizard/TrackedRepoWizard.tsx]: Solid!! 👏🏼 👏🏼

**Pull request merged**